diff --git a/drivers/bluetooth/btqca.c b/drivers/bluetooth/btqca.c
index adcc8f5d..f0dc90bd 100644
--- a/drivers/bluetooth/btqca.c
+++ b/drivers/bluetooth/btqca.c
@@ -58,7 +58,7 @@ static int rome_patch_ver_req(struct hci_dev *hdev, u32 *rome_version)
 	}
 
 	edl = (struct edl_event_hdr *)(skb->data);
-	if (!edl || !edl->data) {
+	if (!edl) {
 		BT_ERR("%s: TLV with no header or no data", hdev->name);
 		err = -EILSEQ;
 		goto out;
@@ -227,7 +227,7 @@ static int rome_tlv_send_segment(struct hci_dev *hdev, int idx, int seg_size,
 	}
 
 	edl = (struct edl_event_hdr *)(skb->data);
-	if (!edl || !edl->data) {
+	if (!edl) {
 		BT_ERR("%s: TLV with no header or no data", hdev->name);
 		err = -EILSEQ;
 		goto out;
